### PR TITLE
fix(tracer): optimizing filter trace

### PR DESCRIPTION
**Pull request by galbash**

### CODE REVIEW

File: src/tracer.js [hunk 1]
```diff
@@ -462,19 +464,35 @@ module.exports.filterTrace = function filterTrace(traceObject, ignoredKeys, remo
         for (let i = 0; i < ignoredKeys.length; i += 1) {
             const predicate = ignoredKeys[i];
             if (typeof predicate === 'string' &&
-            config.processIgnoredKey(predicate) === config.processIgnoredKey(key)) {
+            predicate === config.processIgnoredKey(key)) {
```
Comment (galbash) [inline src/tracer.js:467]: moved the processing of the config to when the config is set (instead of each time)

File: src/config.js [hunk 2]
```diff
@@ -174,8 +177,12 @@ module.exports.setConfig = function setConfig(configData) {
         config.disableHttpResponseBodyCapture = configData.disableHttpResponseBodyCapture;
     }
 
-    if (configData.ignoredKeys) {
-        config.ignoredKeys = configData.ignoredKeys;
+    if (configData.ignoredKeys && Array.isArray(configData.ignoredKeys)) {
+        const filteredIgnoredKeys = configData.ignoredKeys.filter(key => typeof key !== 'function');
```
Comment (amitlicht) [inline src/config.js]: consider filtering explicitly only for strings & regexes, and reject everything else with a warning (not just functions).

File: src/config.js [hunk 3]
```diff
@@ -174,8 +177,12 @@ module.exports.setConfig = function setConfig(configData) {
...
+        if (filteredIgnoredKeys.length !== configData.ignoredKeys.length) {
+            console.warn('Epsagon deprecation warning: function ignoredKeys have been deprecated and will be ignored'); // eslint-disable-line no-console
```
Comment (amitlicht) [inline src/config.js]: I saw there's a `utils.printWarning` function, maybe use that? 
and in any case, add the problematic ignored keys to make this usable.

**Review by amitlicht — CHANGES_REQUESTED** [review]

File: src/config.js (hunk 2 above)
Comment (galbash) [reply src/config.js]: so the problem there is no clear way to filter a regexp (`typeof /wr32/ === 'object'`) - there are some hacks online but they are sketchy at best

File: src/config.js [hunk 4]
```diff
@@ -174,8 +177,15 @@ module.exports.setConfig = function setConfig(configData) {
...
     }
 
-    if (configData.ignoredKeys) {
-        config.ignoredKeys = configData.ignoredKeys;
+    if (configData.ignoredKeys && Array.isArray(configData.ignoredKeys)) {
+        const filteredIgnoredKeys = configData.ignoredKeys.filter(key => typeof key !== 'function');
+        if (filteredIgnoredKeys.length !== configData.ignoredKeys.length) {
+            utils.printWarning(
+                'Epsagon deprecation warning: function ignoredKeys have been deprecated and will be ignored. received ignoredKeys:',
+                configData.ignoredKeys
```
Comment (amitlicht) [inline src/config.js:186]: not critical but I actually meant printing the specific invalid ignored keys.

**Review by amitlicht — APPROVED** [review]

**Pull request merged**